### PR TITLE
FeeCurrencyWhitelist Foundry test

### LINKED ISSUE #10854: test/common/feecurrencywhitelist.ts

### PR DESCRIPTION
**Pull request by pahor167**
### Description

FeeCurrencyWhitelist Foundry test

closes #10854

### CODE REVIEW

File: packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.5.13;
+
+import "celo-foundry/Test.sol";
+import "../../contracts/common/FeeCurrencyWhitelist.sol";
+import "forge-std/console.sol";
```
Comment (soloseng) [inline packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol]: can be removed

File: packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+contract FeeCurrencyWhitelistAddToken is FeeCurrencyWhitelistTest {
+  function testShouldAllowTheOwnerToAddAToken() public {
+    feeCurrencyWhitelist.addToken(address(1));
+    address[] memory whitelist = feeCurrencyWhitelist.getWhitelist();
+    assertEq(whitelist.length, 1);
+    assertEq(whitelist[0], address(1));
+  }
+
+  function testShouldNotAllowNonOwnerToAddAToken() public {
```
Comment (soloseng) [inline packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol]: NIT: To keep with [best practice](https://book.getfoundry.sh/tutorials/best-practices#general-contract-guidance), rename the test to `test_RevertWhen_NonOwnerAddsToken`

File: packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    vm.expectRevert("Ownable: caller is not the owner");
+    vm.prank(nonOwner);
+    feeCurrencyWhitelist.addToken(address(1));
+  }
+}
+
+contract FeeCurrencyWhitelistRemopv is FeeCurrencyWhitelistTest {
```
Comment (soloseng) [inline packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol]:
```suggestion
contract FeeCurrencyWhitelistRemoveToken is FeeCurrencyWhitelistTest {
```

File: packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+  function testShouldRemove() public {
+    feeCurrencyWhitelist.removeToken(address(2), 1);
+    address[] memory whitelist = feeCurrencyWhitelist.getWhitelist();
+    assertEq(whitelist.length, 2);
+    assertEq(whitelist[0], address(1));
+    assertEq(whitelist[1], address(3));
+  }
+
+  function test_ShouldNotRemoveWhenIndexIsWrong() public {
```
Comment (soloseng) [inline packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol]:
```suggestion
  function test_RevertWhen_IndexIsWrong() public {
```

File: packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    vm.expectRevert("Index does not match");
+    feeCurrencyWhitelist.removeToken(address(2), 2);
+  }
+
+  function test_ShouldNotAllowNonOwnerToRemove() public {
```
Comment (soloseng) [inline packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol]:
```suggestion
  function test_RevertWhen_NonOwnerRemovesToken() public {
```

**Review by soloseng — APPROVED** [review]
Looks good! Left a few comments

File: packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol (hunk 2 above)
Comment (pahor167) [reply packages/protocol/test-sol/common/FeeCurrencyWhitelist.sol]: Updated to agreed upon "ShouldRevert_When"

**Pull request merged**